docs/sources/panels/visualizations/geomap.md

```diff
@@ -8,4 +8,123 @@ weight = 600
+
+```ini
+default_baselayer_config = `{
+  "type": "osm-standard",
+  "config": {}
+}`
+```
+
+- **xyz** loads a custom tile server defined by the user.  A valid tile server `url`, with {z}/{x}/{y}, has to be set in order for this option to properly load a default base map.
+
+```ini
+default_baselayer_config = `{
+  "type": "xyz",
+  "config": {
+    "attribution": "Open street map",
+    "url": "https://tile.openstreetmap.org/{z}/{x}/{y}.png"
+  }
+}`
+```
+
+`enable_custom_baselayers` gives the option to enable/disable custom open source base maps that are already implemented. This will be defaulted to `true`.
+
+## Data layer
+
+Data layer in the Geomap plugin determines how you visualize geospatial data on top of the base map.
+
+### Location
+The Geomap panel needs a source of geographical data. The data comes from a database query, and there are four mapping options for your data.
+
+- **Auto** will automatically search for location data. Use this data option when your query is based on one of the following names for data fields.
+  - geohash: “geohash”
+  - latitude: “latitude”, “lat”
+  - longitude: “longitude”, “lng”, “lon”
+  - lookup: “lookup”
+- **Coords** specifies that your query holds coordinate data. You will be prompted to select numeric data fields for latitude and longitude from your database query.
+- **Geohash** specifies that your query holds geohash data. You will be prompted to select a string data field for geohash from your database query.
+- **Lookup** specifies that your query holds location name data that needs to be mapped to a value. You will be prompted to select the lookup field from your database query and a gazetteer. The gazeteer is the directory that will be used to map your queried data to a geographical point.
+
+### Markers Layer
+
+![Markers Layer](/static/img/docs/geomap-panel/geomap_markers_layer.png)
+
+Markers layer allows displaying data points as different marker shapes such as circle, square, triangle, star, and more.
+
+![Markers Layer Options](/static/img/docs/geomap-panel/geomap_markers_options.png)
+
+- **Marker Color** configures the color of the marker. The default `Fixed size` keeps all points a single color. There is an alternate option to have multiple colors depending on the data point values and the threshold set at the `Thresholds` section.
+- **Marker Size** configures the size of the marker. Default is `Fixed size`, making all marker size the same regardless of the data points. However, there is also an option to scale the circles to the corresponding data points. `Min` and `Max` marker size has to be set such that the Marker layer can scale within these range.
+- **Marker Shape** gives flexibility to visualize the data points differently. 
+  - Circle
+  - Square
+  - Triangle
+  - Star
+  - Cross
+  - X
+- **Fill Opacity** configures the transparency of each marker.
+
+### Heatmap Layer
+
+![Heatmap Layer](/static/img/docs/geomap-panel/geomap_heatmap_layer.png)
```

Comment:
Move image.

Reply:
Thank you! Addressed the comments. 

Regarding images in documentations, where should I be adding those images / where is the Grafana/website repo? https://github.com/grafana/grafana/tree/main/docs

Reply:
In the website repo: https://github.com/grafana/website/tree/master/static/static/img/docs, make an folder geomap and add images there. 

Here is useful information on how to name images: https://github.com/grafana/grafana/blob/main/contribute/style-guides/documentation-style-guide.md#images. I encourage users to add a suffix for the version when the image was taken: "image-8-1-0.png".

Reply:
Is this a private repo? I get a page not found with https://github.com/grafana/website/tree/master/static/static/img/docs

Reply:
Not an external repo. If you want, share the images and I can upload them.

Reply:
Hi @achatterjee-grafana We will upload the images here. But why is the docs repo not public? Are the docs generated? This is very confusing for non-Grafana contributors to the project.

Reply:
Tagging @ryantxu, so he can explain why the website repo is not accessible. Ryan, the issue came up as KIm needs to upload images. Thanks.

Reply:
> Not an external repo. If you want, share the images and I can upload them.

geomap-example-8-1-0.png
![geomap-example-8-1-0](https://user-images.githubusercontent.com/43913498/127058989-bd708f3a-60a8-4404-b00e-160f8cd522d7.png)

geomap-baselayer-8-1-0.png
![geomap-baselayer-8-1-0](https://user-images.githubusercontent.com/43913498/127059365-3b9a639c-bf22-4eb9-9d9b-feefe16eb0e6.png)

geomap-markers-8-1-0.png
![geomap-markers-8-1-0](https://user-images.githubusercontent.com/43913498/127060400-4a399558-51c6-4eb4-afbc-dc0b6e63ef6b.png)

geomap-markers-options-8-1-0.png
<img width="411" alt="geomap-markers-options-8-1-0" src="https://user-images.githubusercontent.com/43913498/127059955-d0974909-4acd-4d2b-bf2c-cc189b93dc09.png">

geomap-heatmap-8-1-0.png
![geomap-heatmap-8-1-0](https://user-images.githubusercontent.com/43913498/127060120-f7f08ebb-507e-4278-a1a0-f107b51f47b3.png)

geomap-heatmap-options-8-1-0.png
<img width="409" alt="geomap-heatmap-options-8-1-0" src="https://user-images.githubusercontent.com/43913498/127060185-d2078ff1-db2d-464d-8d96-a0ce2cb7893b.png">